pennylane/vqe/vqe.py

```diff
@@ -0,0 +1,217 @@
+
+                ansatz(params, **kwargs)
+
+            where ``params`` are the trainable weights of the variational circuit, and
+            ``kwargs`` are any additional keyword arguments that need to be passed
+            to the template.
+        device (Device, Sequence[Device]): Corresponding device(s) where the resulting
+            cost function should be executed. This can either be a single device, or a list
+            of devices of length matching the number of terms in the Hamiltonian.
+        interface (str, None): which interface to use.
+            This affects the types of objects that can be passed to/returned to the cost function.
+            Supports all interfaces supported by the :func:`~.qnode` decorator.
+        diff_method (str, None): the method of differentiation to use with the created cost function.
+            Supports all differentiation methods supported by the :func:`~.qnode` decorator.
+
+    Returns:
+        callable: a cost function with signature ``cost_fn(params, **kwargs)`` that evaluates
+        the expectation of the Hamiltonian on the provided device(s).
+
+    .. seealso:: :class:`~.Hamiltonian`, :func:`~.generate_hamiltonian`, :func:`~.map`, :func:`~.dot`
+
+    **Example:**
+
+    First, we create a device and design an ansatz:
+
+    .. code-block:: python
+
+        dev = qml.device('default.qubit', wires=4)
+
+        def ansatz(params, **kwargs):
+            qml.BasisState(np.array([1, 1, 0, 0]), wires=[0, 1, 2, 3])
+            for i in range(4):
+                qml.Rot(*params[i], wires=i)
+            qml.CNOT(wires=[2, 3])
+            qml.CNOT(wires=[2, 0])
+            qml.CNOT(wires=[3, 1])
+
+    Now we can create the Hamiltonian that defines the VQE problem:
+
+    .. code-block:: python3
+
+        coeffs = [0.2, -0.543]
+        obs = [
+            qml.PauliX(0) @ qml.PauliZ(1) @ qml.PauliY(3),
+            qml.PauliZ(0) @ qml.Hadamard(2)
+        ]
+        H = qml.vqe.Hamiltonian(coeffs, obs)
+
+    Alternatively, the :func:`~.generate_hamiltonian` function from the
+    :doc:`/introduction/chemistry` module can be used to generate a molecular
+    Hamiltonian.
+
+    Next, we can define the cost function:
+
+    >>> cost = qml.VQECost(ansatz, hamiltonian, dev, interface="torch")
+    >>> params = torch.rand([4, 3])
+    >>> cost(params)
+    tensor(0.0245, dtype=torch.float64)
+
+    The cost function can be minimized using any gradient descent-based
+    :doc:`optimizer </introduction/optimizers>`.
+    """
+    def __init__(self, ansatz, hamiltonian, device, interface="autograd", diff_method="best"):
+        coeffs, observables = hamiltonian.terms
+        self.hamiltonian = hamiltonian
+        """Hamiltonian: the hamiltonian defining the VQE problem."""
+
+        self.qnodes = qml.map(ansatz, observables, device, interface=interface, diff_method=diff_method)
+        """QNodeCluster: The QNodes to be evaluated. Each QNode corresponds to the
```

Comment:
```suggestion
        """QNodeCollection: The QNodes to be evaluated. Each QNode corresponds to the
```
How did this keep sneaking in?

Reply:
I went back through the git history, a merge conflict caused this.